### PR TITLE
chore(static_obstacle_avoidance, dynamic_obstacle_avoidance): rename avoidance package

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

- Issue: https://github.com/autowarefoundation/autoware/issues/4569
- Discussion: https://github.com/orgs/autowarefoundation/discussions/4741
- Related PR: https://github.com/autowarefoundation/autoware_launch/pull/1008

Rename modules:

- `behavior_path_avoidance_module` -> `autoware_behavior_path_static_obstacle_avoidance_module`
- `behavior_path_dynamic_avoidance_module` -> `autoware_behavior_path_dynamic_obstacle_avoidance_module`

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/53379a7b-8405-5b47-a461-68c8dac47e7b?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_avoidance.param.yaml
Comment (yuki-takagi-66) [inline planning/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_avoidance.param.yaml:1]: Should we change the file name?
I feel the both names are acceptable

**Review by yuki-takagi-66 — APPROVED** [review]
Thank you so much
LGTM
Perhaps the parameter file name should be changed as well, but I don't mind either way.

File: planning/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_avoidance.param.yaml
Comment (satoshi-ota) [reply planning/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_avoidance.param.yaml:1]: @yuki-takagi-66 Thanks. I fixed in https://github.com/autowarefoundation/autoware.universe/pull/7168/commits/ec309e9eee8d2d8d39a68d7dfc163f1d339e104e.

Comment (yuki-takagi-66) [reply planning/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_avoidance.param.yaml:1]: Thank you for your kind attention!

**Review by rej55 — APPROVED** [review]

**Pull request merged**